tests/sentry/utils/test_snuba.py

```diff
@@ -177,6 +179,26 @@ def test_events_dataset_with_project_id(self):
         kwargs, _, _ = _prepare_query_params(query_params)
         assert kwargs["project"] == [self.project.id]
 
+    def test_with_deleted_project(self):
+        query_params = SnubaQueryParams(
+            dataset=Dataset.Events, filter_keys={"project_id": [self.project.id]}
+        )
+
+        self.project.delete()
+        with pytest.raises(UnqualifiedQueryError):
+            get_query_params_to_update_for_projects(query_params)
+
+    @mock.patch("sentry.models.Project.objects.get_from_cache", side_effect=Project.DoesNotExist)
+    def test_with_some_deleted_projects(self, mock_project):
+        other_project = self.create_project(organization=self.organization, slug="a" * 32)
+        query_params = SnubaQueryParams(
+            dataset=Dataset.Events, filter_keys={"project_id": [self.project.id]}
```

Comment:
Shouldn't this have `other_project.id` as a filter too?

Reply:
dang good catch :+1: